webroot/js/directives/sentence-and-translations.dir.js

```diff
@@ -146,16 +147,53 @@
             resetLists();
         });
 
+        $scope.$on('randomSentenceRequested', function(event, lang) {
+            var url = rootUrl + '/sentences/random';
+            if (lang) {
+                url += '/' + lang;
+            }
+            vm.inProgress = true;
+            $http.get(url).then(function(result) {
+                if (!result.data || !result.data.random) {
+                    // TODO Show error
+                    return;
+                }
```

Comment:
This won’t work when the returned HTTP code is 4xx or 5xx. You have to do it like this:
```js
$http.get(url).then(
    function(response) {
        // Server did handle the request normally
        // We got whatever data has been set by the action method
    },
    function(response) {
        // server couldn’t handle the request (not logged in,
        // unauthorized, maintenance ongoing, ddos ongoing...)
    }
).finally(function() {
    // that gets additionally executed in the end no matter what
});
```
By the way, if you don’t provide the second parameter to `.then()`, a warning is printed in the browser console.

Reply:
I would keep this for some refactoring session. This is an issue for every $http request. We should set up some generic error handling for status 4xx and 5xx.